packages/ai-chat/src/common/chat-model.ts

```diff
@@ -420,21 +512,66 @@ export class ChatModelImpl implements ChatModel {
     }
 }
 
+export class ChangeSetImpl implements ChangeSet {
+    protected readonly _onDidChangeEmitter = new Emitter<void>();
+    onDidChange: Event<void> = this._onDidChangeEmitter.event;
+
+    protected _elements: ChangeSetElement[] = [];
+
+    constructor(public readonly title: string, elements: ChangeSetElement[] = []) {
+        this.addElements(elements);
+    }
+
+    getElements(): ChangeSetElement[] {
+        return this._elements;
+    }
+
+    addElement(element: ChangeSetElement): void {
+        this.addElements([element]);
```

Comment:
Why is no notification triggered in this case?

Reply:
this just delegates to the addElements method below, which triggers the notification